tests/handlers/auth_test.py

```diff
@@ -0,0 +1,358 @@
+"""Tests for the /auth route."""
+
+from __future__ import annotations
+
+import base64
+import os
+import time
+from typing import TYPE_CHECKING
+from unittest.mock import ANY
+
+import jwt
+
+from jwt_authorizer.config import ALGORITHM
+from jwt_authorizer.session import SessionStore, Ticket
+from tests.util import RSAKeyPair, create_test_app, create_test_token
+
+if TYPE_CHECKING:
+    from aiohttp.pytest_plugin.test_utils import TestClient
+
+
+def assert_www_authenticate_header_matches(
+    header: str, method: str, error: str
+) -> None:
+    header_method, header_info = header.split(" ", 1)
+    assert header_method == method
+    if header_method == "Basic":
+        assert header_info == 'realm="tokens"'
+    else:
+        data = header_info.split(",")
+        assert data[0] == 'realm="tokens"'
+        assert data[1] == f'error="{error}"'
+        assert data[2].startswith("error_description=")
+
+
+async def test_authnz_token_no_auth(aiohttp_client: TestClient) -> None:
+    app = await create_test_app()
+    client = await aiohttp_client(app)
+
+    r = await client.get("/auth", params={"capability": "exec:admin"})
+    assert r.status == 401
+    assert r.headers["WWW-Authenticate"]
+    assert_www_authenticate_header_matches(
+        r.headers["WWW-Authenticate"], "Bearer", "No Authorization header"
+    )
+
+    r = await client.get(
+        "/auth",
+        params={"capability": "exec:admin"},
+        headers={"Authorization": "Bearer"},
+    )
+    assert r.status == 401
+    assert r.headers["WWW-Authenticate"]
+    assert_www_authenticate_header_matches(
+        r.headers["WWW-Authenticate"], "Bearer", "Unable to find token"
```

Comment:
is this a different case than passing in an invalid token?

Reply:
Yeah, I want to make sure that failing to have a token at all didn't cause some 500 error inside the code. Python is somewhat vulnerable to that.